broker/src/test/java/io/camunda/zeebe/broker/partitioning/PartitionLeaveTest.java

```diff
@@ -83,6 +85,128 @@ void canStillProcessAfterLeaving(@TempDir final Path tmp) {
     }
   }
 
+  @Test
+  void shouldRemoveDataAfterLeaving(@TempDir final Path tmp) {
+    // given -- two brokers replicating partition 1
+    final var broker0 =
+        buildBroker(
+            tmp.resolve("broker-0"),
+            brokerCfg -> {
+              final var clusterCfg = brokerCfg.getCluster();
+              clusterCfg.setClusterSize(2);
+              clusterCfg.setNodeId(0);
+              clusterCfg.setPartitionsCount(2);
+              clusterCfg.setReplicationFactor(2);
+            });
+    final var initialContactPoint =
+        broker0.getConfig().getNetwork().getInternalApi().getAdvertisedAddress();
+
+    final var broker1 =
+        buildBroker(
+            tmp.resolve("broker-1"),
+            brokerCfg -> {
+              final var clusterCfg = brokerCfg.getCluster();
+              clusterCfg.setInitialContactPoints(
+                  List.of(initialContactPoint.getHostName() + ":" + initialContactPoint.getPort()));
+
+              clusterCfg.setClusterSize(2);
+              clusterCfg.setNodeId(1);
+              clusterCfg.setPartitionsCount(2);
+              clusterCfg.setReplicationFactor(2);
+            });
+    CompletableFuture.allOf(
+            CompletableFuture.runAsync(broker0::start), CompletableFuture.runAsync(broker1::start))
+        .join();
+
+    try (final var client =
+        ZeebeClient.newClientBuilder()
+            .usePlaintext()
+            .gatewayAddress("localhost:" + broker0.getConfig().getGateway().getNetwork().getPort())
+            .build()) {
+      Awaitility.await()
+          .untilAsserted(
+              () ->
+                  TopologyAssert.assertThat(client.newTopologyRequest().send().join())
+                      .isComplete(2, 2, 2));
+
+      // when -- one broker leaves partition 1
+      ((PartitionManagerImpl) broker1.getBrokerContext().getPartitionManager()).leave(1).join();
```

Comment:
:thought_balloon: This hints that we should probably make this a QA test...but then, the fact we look at the data path hints we shouldn't. Hmmm, decisions decisions...